tests/queries/0_stateless/02504_regexp_dictionary_table_source.sql

```diff
@@ -0,0 +1,81 @@
+-- Tags: use-vectorscan
+
+DROP TABLE IF EXISTS regexp_dictionary_source_table;
+DROP DICTIONARY IF EXISTS regexp_dict1;
+
+CREATE TABLE regexp_dictionary_source_table
+(
+    id UInt64,
+    parent_id UInt64,
+    regexp String,
+    keys   Array(String),
+    values Array(String),
+) ENGINE=TinyLog;
+
+-- test back reference.
+
+INSERT INTO regexp_dictionary_source_table VALUES (1, 0, 'Linux/(\d+[\.\d]*).+tlinux', ['name', 'version'], ['TencentOS', '\1'])
+INSERT INTO regexp_dictionary_source_table VALUES (2, 0, '(\d+)/tclwebkit(\d+[\.\d]*)', ['name', 'version', 'comment'], ['Andriod', '$1', 'test $1 and $2'])
+INSERT INTO regexp_dictionary_source_table VALUES (3, 2, '33/tclwebkit', ['version'], ['13'])
+INSERT INTO regexp_dictionary_source_table VALUES (4, 2, '3[12]/tclwebkit', ['version'], ['12'])
+INSERT INTO regexp_dictionary_source_table VALUES (5, 2, '3[12]/tclwebkit', ['version'], ['11'])
+INSERT INTO regexp_dictionary_source_table VALUES (6, 2, '3[12]/tclwebkit', ['version'], ['10'])
+
+create dictionary regexp_dict1
+(
+    regexp String,
+    name String,
+    version Nullable(UInt64),
+    comment String default 'nothing'
+)
+PRIMARY KEY(regexp)
+SOURCE(CLICKHOUSE(QUERY concat('select * from ', currentDatabase() , '.regexp_dictionary_source_table')))
```

Comment:
Why it's not just `SOURCE(CLICKHOUSE(TABLE 'regexp_dictionary_source_table'))`?

Reply:
because the schema will mismatch if we refer a TABLE. Though if we refer a QUERY, the schema will be decided by sample_block. That's trikcy but I think it's ok for test.